core/error.go

```diff
@@ -177,3 +177,20 @@ func IsValueTransferErr(e error) bool {
 	_, ok := e.(*ValueTransferError)
 	return ok
 }
+
+type BadHashErr struct {
+	Message string
+}
+
+func (err *BadHashErr) Error() string {
+	return err.Message
+}
+
+func BadHashError(hash common.Hash) error {
+	return &BadHashErr{Message: fmt.Sprintf("Found known bad hash in chain %x", hash)}
+}
```

Comment:
Don't create a custom constructor for error messages. Instead, do the custom message formatting within the `Error()` method as in http://golang.org/src/net/net.go?s=15964:15997#L519.